CSharp/JobPrep/Program.cs

```diff
@@ -0,0 +1,333 @@
+﻿#define VERBOSE
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Concurrent;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Microsoft.Azure.Batch;
+using Microsoft.Azure.Batch.Auth;
+using Microsoft.Azure.Batch.Common;
+
+
+namespace JobPrep
+{
+    public class Program
+    {
+        private const string Url = "https://batch.core.windows.net";
+
+        // insert your batch account name and key along with the name of a pool to use. If the pool is new or has no VMs, 3
+        // VMs will be added to the pool to perform work.
+        private static string BatchAccount = "<batch_account>";
+        private static string BatchKey = "<batch_key>";
+        private const string PoolName = "HelloWorld-Pool";
+
+        // similarly, you need a storage account for the file staging example
+        private const string StorageAccount = "<storage_account>";
+        private const string StorageKey = "<storage_key>";
+        private const string StorageBlobEndpoint = "https://" + StorageAccount + ".blob.core.windows.net";
+
+        public static void Main(string[] args)
+        {
+            // This will boost parallel submission speed for REST APIs. If your use requires many simultaneous service calls set this number to something large, such as 100.  
+            // See: http://msdn.microsoft.com/en-us/library/system.net.servicepointmanager.defaultconnectionlimit%28v=vs.110%29.aspx for more info.
+            System.Net.ServicePointManager.DefaultConnectionLimit = 20;
+
+            if (File.Exists("..\\..\\batchcred.txt"))
```

Comment:
this is a new way to use creds with Batch compared to all the other samples. I'd rather we have one way of doing this which is consolidated into a common assembly as I mentioned earlier. Pool creation/retention policy should be there as well.